controllers/gce/loadbalancers/loadbalancers.go

```diff
@@ -834,65 +880,52 @@ func mapsEqual(a, b *compute.UrlMap) bool {
 func (l *L7) Cleanup() error {
 	if l.fw != nil {
 		glog.Infof("Deleting global forwarding rule %v", l.fw.Name)
-		if err := l.cloud.DeleteGlobalForwardingRule(l.fw.Name); err != nil {
-			if !utils.IsHTTPErrorCode(err, http.StatusNotFound) {
-				return err
-			}
+		if err := utils.IgnoreHTTPNotFound(l.cloud.DeleteGlobalForwardingRule(l.fw.Name)); err != nil {
+			return err
 		}
 		l.fw = nil
 	}
 	if l.fws != nil {
 		glog.Infof("Deleting global forwarding rule %v", l.fws.Name)
-		if err := l.cloud.DeleteGlobalForwardingRule(l.fws.Name); err != nil {
-			if !utils.IsHTTPErrorCode(err, http.StatusNotFound) {
-				return err
-			}
-			l.fws = nil
+		if err := utils.IgnoreHTTPNotFound(l.cloud.DeleteGlobalForwardingRule(l.fws.Name)); err != nil {
+			return err
 		}
+		l.fws = nil
 	}
 	if l.ip != nil {
-		glog.Infof("Deleting static IP %v(%v)", l.ip.Name, l.ip.Address)
-		if err := l.cloud.DeleteGlobalStaticIP(l.ip.Name); err != nil {
-			if !utils.IsHTTPErrorCode(err, http.StatusNotFound) {
-				return err
-			}
-			l.ip = nil
+		glog.V(2).Infof("Deleting static IP %v(%v)", l.ip.Name, l.ip.Address)
+		// Ignore err when deleting static IP since multiple ingresses could be sharing one
+		if err := utils.IgnoreHTTPNotFound(l.cloud.DeleteGlobalStaticIP(l.ip.Name)); err != nil {
+			glog.Warningf("Failed to delete global static ip %v (%v), but will continue with remaining cleanup. err: %v", l.ip.Name, l.ip.Address, err)
 		}
+		l.ip = nil
```

Comment:
`l.ip` is a bit special since it can be a shared resource among ingresses. We want to continue other resource cleanup even if this fails.